### PR TITLE
Check-in point for the template extension

### PR DESCRIPTION
**Pull request by jamarw**
1. Add misc extension files (ignore, settings, etc.).
2. Add docs.template settings to package.json and chai dependency for testing.
3. Add setting configuration check to extension.ts.
4. Break-up template controller into smaller, resuable functions for future functionality i.e. validation.
5. Add metadata replace functionality to template controller.
6. Remove unused code from common.
7. Create github helper for repo functions.
8. Create user-metadata helper to manage metadata information.

### CODE REVIEW

File: docs-article-templates/src/helper/common.ts [hunk 1]
```diff
@@ -105,137 +42,19 @@ export function hasValidWorkSpaceRootPath(senderName: string) {
...
  */
 export function debug(message: any) {
     process.stdout.write(message + "\n");
 }
+
+export function generateTimestamp() {
+    const date = new Date(Date.now());
+    const currentYear = date.getFullYear();
+    // In Javascript, the month starts from 0 to 11, so we must add 1 to get the current month
+    const currentMonth = (date.getMonth() + 1);
```
Comment (wibjorn) [inline docs-article-templates/src/helper/common.ts]:
```
     (
             ( ,)
            ). ( )
           (, )' (.
          \WWWWWWWW/
           '--..--'
              }{
              {}
```

Comment (wibjorn) [reply docs-article-templates/src/helper/common.ts]: dates rule

File: docs-article-templates/src/helper/github.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+"use-strict";
+
+import * as fs from "fs";
+import os = require("os");
+import * as path from "path";
+import * as vscode from "vscode";
+import * as common from "./common";
+
+// to-do: change the console output to another form of user notification prior to production release.
+/* tslint:disable:no-console */
```
Comment (wibjorn) [inline docs-article-templates/src/helper/github.ts]: super minor, might consider in the future just oneline turnoffs so, if you're like me, you don't accidentally add console where you don't want them

**Comment by wibjorn** [comment]
think your package.json needs a dark icon option

```
 {
        "command": "markdownQuickPick",
        "title": "Docs Markdown Authoring",
        "icon": {
          "dark": "images/svg/markdown-light.svg",
          "light": "images/svg/markdown-dark.svg"
        }
      }
```

File: docs-article-templates/src/helper/github.ts (hunk 2 above)
Comment (jamarw) [reply docs-article-templates/src/helper/github.ts]: Good point

**Comment by jamarw** [comment]
@wibjorn Great catch on the themes and icons.  Added two new icons and updated the references in package.json.

**Pull request merged**